aten/src/ATen/native/AutogradComposite.cpp

```diff
@@ -22,6 +23,46 @@ std::tuple<at::Tensor, at::Tensor> _unpack_dual(const at::Tensor& tensor, int64_
   return std::tuple<at::Tensor, at::Tensor>(tensor._fw_primal(level), tensor._fw_grad(level));
 }
 
+Tensor _new_zeros_with_same_feature_meta(
+    const at::Tensor& self,
+    const at::Tensor& other,
+    int64_t self_num_batch_dims) {
+  // This implementation only applies to the batched grad case
+  auto other_sizes = other.sizes();
+  auto other_strides = other.strides();
+  auto other_storage_offset = other.storage_offset();
+
+  auto self_sizes = self.sizes();
+  auto self_strides = self.strides();
+  auto self_num_feature_dims = self.dim() - self_num_batch_dims;
+
+  // NB: We don't check that the sizes of self is the same as that of other
+  //     because this function is also used in the inplace over view case
+
+  std::vector<int64_t> out_sizes;
+  out_sizes.reserve(self.dim());
+  out_sizes.insert(out_sizes.begin(), other_sizes.begin(), other_sizes.end());
+  out_sizes.insert(out_sizes.begin(), self_sizes.begin(), self_sizes.end() - self_num_feature_dims);
+
+  // We use the strides of other, and tack on the strides computed with
+  // the batch dims of self, so that the slices are arranged contiguously
+  std::vector<int64_t> out_strides;
+  out_strides.reserve(self.dim());
+  out_strides.insert(out_strides.begin(), other_strides.begin(), other_strides.end());
+
+  int64_t prod = other.storage().nbytes() / other.itemsize();
```

Comment:
as_strided should now work for forward AD (tested locally)
However, because it uses .storage(), the main potential concern is supporting hessian computation in functorch which does vmap(jvp(vmap(vjp, where we have a batched tensor wrapping a TensorWrapper